packages/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -593,7 +593,7 @@ export const GridRootStyles = styled('div', {
         borderTop: 'none',
       },
     },
-    [`&.${c['root--disableUserSelection']} .${c.cell}`]: {
+    [`&.${c['root--disableUserSelection']}`]: {
```

Comment:
Do we really need a class for this? It feels more verbose than it needs to be, we could just set `style.userSelect = '...'` directly.